src/back/TicketManager/src/PaymentService/TicketManager.PaymentService.Domain/Payments/Payment.cs

```diff
@@ -7,15 +7,47 @@ public class Payment: IAggregateRoot<Guid>, IOptimisticConcurrent
     public Guid Id { get; private init; }
     public PaymentStatus PaymentStatus { get; private set; }
     public DateTime DateCreated { get; private init; }
+    public DateTime DateToExpire { get; private init; }
+    private const int ExpirationTimeInMinutes = 30;
```

Comment:
You don't have to store `DateToExpire`.

You can just create property like
`public bool IsExpired => DateCreated + TimeToExpire <= DateTime.UtcNow;`